src/App.vue

```diff
@@ -1,7 +1,14 @@
 <template>
   <main>
-    <c-header />
-    <c-table :messages="messages" />
+    <c-header
+      @onUpdateMessages="updateMessages"
+      @onUpdateRefreshInterval="updateRefreshInterval"
+      :refreshInterval="refreshInterval"
+    />
+    <c-table :messages="messages" class="pt-32" />
+    <div class="flex w-full">
+      <div class="loader" v-if="isLoading">Loading...</div>
```

Comment:
The thing about the spinner is that it's weird to see it on the screen for less 50ms or something like this. What we do in other projects at wiremind is that we only display the spinner after 150ms and we make it visible for at least 300ms. It may be a little complex and not prioritary but could you add an issue about that ?